packages/synthetix-main/contracts/modules/AccountModule.sol

```diff
@@ -0,0 +1,71 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@synthetixio/core-contracts/contracts/ownership/OwnableMixin.sol";
+import "@synthetixio/core-contracts/contracts/proxy/UUPSProxy.sol";
+import "@synthetixio/core-contracts/contracts/initializable/InitializableMixin.sol";
+import "@synthetixio/core-contracts/contracts/satellite/SatelliteFactory.sol";
+import "../interfaces/IAccountModule.sol";
+import "../storage/AccountModuleStorage.sol";
+
+import "../satellites/Account.sol";
+
+contract AccountModule is IAccountModule, OwnableMixin, AccountModuleStorage, InitializableMixin, SatelliteFactory {
+    event AccountCreated(address accountAddress);
+
+    function _isInitialized() internal view override returns (bool) {
+        return _accountModuleStore().initialized;
+    }
+
+    function isAccountModuleInitialized() external view override returns (bool) {
+        return _isInitialized();
+    }
+
+    function initializeAccountModule() external override onlyOwner onlyIfNotInitialized {
+        AccountModuleStore storage store = _accountModuleStore();
+
+        Account firstAccountImplementation = new Account();
```

Comment:
will there ever be anything beyond the "first account" implementation?

Reply:
yes, the AccountModule is the "implementation" manager. And allows the system to be upgraded with another implementation of Account contract